app/controllers/criteria_controller.rb

```diff
@@ -224,8 +224,16 @@ def rubric_criterion_params
                                              :position,
                                              :ta_visible,
                                              :peer_visible,
+                                             :max_mark,
+                                             levels_attributes: [:id, :name, :mark, :description],
                                              assignment_files: []).to_h.deep_merge(params.require(:rubric_criterion)
-                                                                           .permit(:max_mark).to_h)
+                                                                           .permit(:max_mark,
```

Comment:
Can you try deleting the `to_h` and everything after it on the previous line? I'm not sure this entire section is still necessary.